### DISCUSSION TITLE
Is there an option for ALEImport to use relative paths instead of starting from project root?

### DISCUSSION THREAD

**Discussion by ryabrody** [w0rp/ale#4364] [Q&A - Ask for help with problems]
ale provides the command `:ALEImport` which tries to find an import for the symbol at the cursor and inserts the first one it finds. The import is inserted to the top of the file. The import path starts at the root directory of the application. Example:
```
import { ExampleException } from "src/exceptions/excample.exception";
```

I would like to configure that the path to the module file is relative from the current file and also that single quotes instead of double quotes are used:
```
import { ExampleException } from '../exceptions/excample.exception';
```

I searched the documentation if its possible to add some configurations to `:ALEImport `command but i did not found anything regarding this. I would like to avoid installing another plugin but it seems not to be supported by ale itself.

**Comment by hsanson** [comment]
ALE simply imports whatever the underlying LSP returns. If the LSP has such configuration then it should be set on the LSP.

Also ALE always selects the first import returned by LSP even if it returns multiple options. There is an issue to address this but I do not see it solved any time soon: https://github.com/dense-analysis/ale/issues/4085

**Reply by ryabrody** [reply]
@hsanson 
Thanks for your reply. Do you may know how i could set it on the LSP / Change the configuration in LSP? I am not familiar with the LSP. Just using ale with vim at the moment.

**Reply by hsanson** [reply]
Sorry, I do not know what LSP you are trying to use.  Post the output of ALEInfo and maybe someone using the same LSP you have enabled can help.

**Reply by ryabrody** [reply]
I think i am using the tsserver lsp with ale as lsp client. ALEInfo output:
```
:ALEInfo
 Current Filetype: typescript
Available Linters: ['deno', 'eslint', 'standard', 'tslint', 'tsserver', 'typecheck', 'xo']
  Enabled Linters: ['eslint', 'tsserver', 'typecheck']
  Ignored Linters: []
 Suggested Fixers:
  'deno' - Fix TypeScript using deno fmt.
  'eslint' - Apply eslint --fix to a file.
  'prettier' - Apply prettier to a file.
  'remove_trailing_lines' - Remove all blank lines at the end of a file.
  'trim_whitespace' - Remove all trailing whitespace characters at the end of every line.
  'tslint' - Fix typescript files with tslint --fix.
  'xo' - Fix JavaScript/TypeScript files using xo --fix.
 Linter Variables:
let g:ale_typescript_tsserver_config_path = ''
let g:ale_typescript_tsserver_executable = 'tsserver'
let g:ale_typescript_tsserver_use_global = 0
 Global Variables:
let g:ale_cache_executable_check_failures = v:null
let g:ale_change_sign_column_color = 0
let g:ale_command_wrapper = ''
let g:ale_completion_delay = v:null
let g:ale_completion_enabled = 0
let g:ale_completion_max_suggestions = v:null
let g:ale_disable_lsp = 0
let g:ale_echo_cursor = 1
let g:ale_echo_msg_error_str = 'Error'
let g:ale_echo_msg_format = '%code: %%s'
let g:ale_echo_msg_info_str = 'Info'
let g:ale_echo_msg_warning_str = 'Warning'
let g:ale_enabled = 1
let g:ale_fix_on_save = 0
let g:ale_fixers = {}
let g:ale_history_enabled = 1
let g:ale_history_log_output = 1
let g:ale_keep_list_window_open = 0
let g:ale_lint_delay = 200
let g:ale_lint_on_enter = 1
let g:ale_lint_on_filetype_changed = 1
let g:ale_lint_on_insert_leave = 1
let g:ale_lint_on_save = 1
let g:ale_lint_on_text_changed = 'normal'
let g:ale_linter_aliases = {}
let g:ale_linters = {'typescript': ['eslint', 'tsserver', 'typecheck']}
let g:ale_linters_explicit = 0
let g:ale_linters_ignore = {}
let g:ale_list_vertical = 0
let g:ale_list_window_size = 10
let g:ale_loclist_msg_format = '%code: %%s'
let g:ale_max_buffer_history_size = 20
let g:ale_max_signs = -1
let g:ale_maximum_file_size = v:null
let g:ale_open_list = 0
let g:ale_pattern_options = v:null
let g:ale_pattern_options_enabled = v:null
let g:ale_root = {}
let g:ale_set_balloons = 0
let g:ale_set_highlights = 1
let g:ale_set_loclist = 1
let g:ale_set_quickfix = 0
let g:ale_set_signs = 1
let g:ale_sign_column_always = 0
let g:ale_sign_error = '>>'
let g:ale_sign_info = '--'
let g:ale_sign_offset = 1000000
let g:ale_sign_style_error = '>>'
let g:ale_sign_style_warning = '--'
let g:ale_sign_warning = '--'
let g:ale_sign_highlight_linenrs = 0
let g:ale_statusline_format = v:null
let g:ale_type_map = {}
let g:ale_use_global_executables = v:null
let g:ale_virtualtext_cursor = 0
let g:ale_warn_about_trailing_blank_lines = 1
let g:ale_warn_about_trailing_whitespace = 1

```

**Reply by hsanson** [reply]
Typescript, javascript, and anything on that front is outside my area of expertise.  You can check tsserver documentation to see how to configure it but from what I saw there is no way to change how imports are returned by it:

 - https://github.com/microsoft/TypeScript/wiki/Standalone-Server-%28tsserver%29

**Reply by ryabrody** [reply]
I think it would be `importModuleSpecifierPreference` (https://github.com/microsoft/TypeScript/blob/main/lib/tsserverlibrary.d.ts#L8358) to be set to `relative` but i also do not see a guide on how or where this can be set.

**Comment by ryabrody** [comment]
For typescript/tsserver, I now was able to switch to relative import paths by removing the `"baseUrl": "./",` from the `tsconfig.json` file:
https://www.typescriptlang.org/tsconfig#baseUrl
